### PR TITLE
Add External Immich URL Redirection via Environment Variable

### PR DESCRIPTION
**Pull request by sai80082**
- Added a new environment variable `EXTERNAL_IMMICH_URL` to handle redirection to an external Immich URL when provided, instead of using the internal link. This ensures proper routing for external access.

### CODE REVIEW

**Comment by Tyree** [comment]
I assume if you're already using the external address, this will not affect that setting. Or should I instead use this new variable for the external address?
My immich setup requires the https protocol, so I have to use the external address I set with my reverse proxy. (I just have my router redirect all local requests for that address back to my internal server.)

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -29,6 +30,7 @@ const App = ({ Component, pageProps, ...props }: AppPropsWithProps) => {
 App.getInitialProps = async () => {
   return {
     props: {
+      exImmichUrl: ENV.EXTERNAL_IMMICH_URL,
```
Comment (varun-raj) [inline src/pages/_app.tsx:33]: Actually we wont use `immichURL` if its internal in the frontend, we can directly replace this key with `ENV.EXTERNAL_IMMICH_URL`

**Review by varun-raj — CHANGES_REQUESTED** [review]

File: src/pages/_app.tsx (hunk 1 above)
Comment (sai80082) [reply src/pages/_app.tsx:33]: yeah thanks will be fixed.

Comment (sai80082) [reply src/pages/_app.tsx:33]: Hey i think we should keep immichurl as we are using it in the connection info section
```
 <p className="text-muted-foreground font-mono">Connected to (Internal)</p>
        <Link href={immichURL} target="_blank" className="font-mono">
          {immichURL}
        </Link>
```

**Pull request merged**